sos/report/plugins/ds.py

```diff
@@ -43,11 +45,20 @@ def setup(self):
             "/etc/dirsrv/admin-serv/password.conf"
         ])
 
+        # check for dsctl, and if it has the healthcheck option
+        HAVE_HEALTHCHECK = False
+        if os.path.exists('/usr/sbin/dsctl'):
+            dsctl_output = subprocess.check_output(['dsctl', '--help'])
+            if 'healthcheck' in str(dsctl_output):
+                HAVE_HEALTHCHECK = True
+
         try:
             for d in os.listdir("/etc/dirsrv"):
                 if d[0:5] == 'slapd':
                     certpath = os.path.join("/etc/dirsrv", d)
                     self.add_cmd_output("certutil -L -d %s" % certpath)
+                    if HAVE_HEALTHCHECK:
```

Comment:
You can use `SoSPredicate` rather than calling `subprocess` methods. See e.g.:

https://github.com/sosreport/sos/blob/master/sos/report/plugins/lvm2.py#L54-L55

(you need to import `SoSPredicate` from `sosreport.plugins`)

Reply:
OK, changes made, tests pass, please review...